module/zfs/arc.c

```diff
@@ -4189,7 +4233,15 @@ arc_evict_impl(arc_state_t *state, arc_buf_contents_t type, int64_t bytes)
 	if (bytes > 0 && zfs_refcount_count(&state->arcs_esize[type]) > 0) {
 		delta = MIN(zfs_refcount_count(&state->arcs_esize[type]),
 		    bytes);
-		return (arc_evict_state(state, type, 0, delta));
+		/*
+		 * Create a list of guids from any active ARC async flushes.
+		 * The underlying arc_evict_state() function will target
+		 * only spa guids from this list when it is not empty.
+		 */
+		uint64_t spa_list[16];
+		unsigned int spa_cnt =
+		    arc_async_flush_init_spa_list(spa_list, 16);
```

Comment:
`arc_evict_impl()` is used not only for flushing, but also for regular evictions.  I can see some logic in a guess that if we need to flush something, eviction of other data same time may be less productive.  But I am not sure we can know that particular state we are asked to evict now has data for the pools being flushed.  Any way, since we are already actively flushing, memory pressure is quite unlikely, so I don't think we should get here often.

Also hard-coded array of 16 entries looks dirty.

Reply:
I abandoned the commit with this change (Changed arc evict to prioritize unloaded spas)